src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -88,6 +90,7 @@ private sealed class LatestDiagnosticsForSpanGetter
                  Func<string, bool>? shouldIncludeDiagnostic,
                  CancellationToken cancellationToken)
             {
+                var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
this is the pull-diags entrypoint for computing diagnostics for a document.  we'll always need to get the tree/text for it.  So getting the text here should not be problem